### DISCUSSION TITLE
help/minimum-reproducible-example/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#19445] [Docs]
# help/minimum-reproducible-example/

Learn how to create effective Minimum Reproducible Examples (MRE) for bug reports in Ultralytics YOLO repositories. Follow our guide for efficient issue resolution.

https://docs.ultralytics.com/help/minimum-reproducible-example/

**Comment by AVON257** [comment]
Hi, I'd just like to mention that I have found a bug on the validation model (if it is a bug), where when I train a nano sized OBB model on YOLO 12 unfortunately it either produces Nan's during training or it won't validate on my RTX3090, even though it isn't showing that it's using too much memory when I try it with any batch size.

Any ideas on this? 

Thanks,

Tom T.

P.S. I am actively working on combining this with quantum circuits but in order to do that, I need to be able to test it against the classical baseline, and curiously it seems okay on Colab when I tried it on an A100 but not on my RTX3090.

P.P.S. I would very much like to submit a commit to you once I finish my current work on the latest version of Ultralytics if you don't mind, although I should tell you that it will only work with optimal performance on Nvidia hardware at the moment. It can work on other things too, it just won't work as fast that way.

**Reply by AVON257** [reply]
In fact I would debug it myself if I could figure out which part of ultralytics deals with validation, any ideas?

**Reply by glenn-jocher** [reply]
@AVON257 hi Tom, thanks for reporting this. To help us investigate, please create a Minimum Reproducible Example using the COCO8-OBB dataset and share your exact training/validation commands, dependency versions (`yolo checks`), and complete error logs. The validation logic is handled in the Val mode - you can review the source code in the Ultralytics repository. 

For potential contributions, we welcome PRs that follow our Contributing Guide including signed CLA and Google-style docstrings. The community will appreciate your improvements to YOLO compatibility across hardware platforms.

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for sharing this with the Ultralytics community 🚀! We appreciate your interest and contribution.

We recommend taking a look at the [Docs](https://docs.ultralytics.com/) for guidance, including our detailed instructions on creating a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to ensure efficient issue resolution. Providing a clear MRE helps us and the community quickly identify and resolve any reported issues.

If this is a 🐛 Bug Report, please share a complete MRE or additional details about your environment (e.g., Python and PyTorch versions) that can help us replicate and debug the issue effectively. 

If you have a ❓ Question about custom training, make sure to include as much context as possible, such as your dataset setup, training logs, and the steps you’ve tried. You can also refer to our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to optimize your training process.

Join the broader Ultralytics community for discussions:
- For real-time conversations, visit our [Discord](https://discord.com/invite/ultralytics) 🎧.
- For detailed threads, check out our [Discourse](https://community.ultralytics.com/).
- Or explore knowledge-sharing on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Ensure your `ultralytics` package is up-to-date to verify your issue hasn't already been resolved. Upgrade with the latest [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of the supported environments below, with dependencies like [CUDA](https://developer.nvidia.com/cuda-zone), [CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

Stay informed with the system's CI status! If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests have passed successfully, ensuring verified functionality across macOS, Windows, and Ubuntu.

This is an automated response 🤖, but don’t worry! An Ultralytics engineer will assist you soon to ensure your experience is as seamless as possible. 😊

**Comment by AVON257** [comment]
Thanks for this, I'll get back to you as soon as I can, although it was the
lvis dataset that produced these errors.

Does that make any difference?

**Reply by glenn-jocher** [reply]
@AVON257 hi there! While the LVIS dataset context is helpful, we still need a **Minimum Reproducible Example** with the **COCO8-OBB dataset** (or another public dataset) to efficiently investigate this issue. Please share your exact training/validation commands, `yolo checks` output, and full error traceback when using a public dataset. This standardized approach helps us resolve issues faster for everyone. For MRE guidelines, see [Creating a Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum-reproducible-example/). 😊
